pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -671,5 +694,40 @@ func (p *criStatsProvider) getContainerLogStats(path string, rootFsInfo *cadviso
 	result.UsedBytes = &usedbytes
 	inodesUsed := uint64(logMetrics.InodesUsed.Value())
 	result.InodesUsed = &inodesUsed
-	return result
+	result.Time = maxUpdateTime(&result.Time, &logMetrics.Time)
+	return result, nil
+}
+
+// getPodLogStats gets stats for logs under the pod log directory. Container logs usually exist
+// under the container log directory. However, for some container runtimes, e.g. kata, gvisor,
+// they may want to keep some pod level logs, in that case they can put those logs directly under
+// the pod log directory. And kubelet will take those logs into account as part of pod ephemeral
+// storage.
+func (p *criStatsProvider) getPodLogStats(path string, rootFsInfo *cadvisorapiv2.FsInfo) (*statsapi.FsStats, error) {
```

Comment:
what is the `rootFsInfo` passed in, and why do we add it to the log usage?  It seems like this function returns aggregate pod-level FsStats, rather than just stats for pod logs.  Or maybe it is missing volume stats?  It would be better to either have the function return aggregate pod stats, or just log stats and do the merging with other stats outside of this function.

Reply:
This only returns pod log stats, basically the stats of all logs under `/var/log/pods/NAMESPACE_NAME_UID/*.log`.

The rootFsInfo is used to set the available and capacity, we use it in this way everywhere I think.

Reply:
oh, that makes total sense